### PR TITLE
Compile switch for different handheld contact switches & record greeting on the phone itself & extensive documentation

### PR DESCRIPTION
**Pull request by DD4WH**

### CODE REVIEW

File: audio-guestbook.ino [hunk 1]
```diff
@@ -38,6 +44,10 @@
 #define HOOK_PIN 0
 #define PLAYBACK_BUTTON_PIN 1
 
+// comment his out, if your handheld OPENS the contact on lift
```
Comment (robinvan0418) [inline audio-guestbook.ino:47]: Found a typo here :)

Comment (DD4WH) [reply audio-guestbook.ino]: Thanks!
Corrected and added a lot more configuration options and made the sketch much easier to use for the beginner.

File: audio-guestbook.ino [hunk 2]
```diff
@@ -27,16 +34,46 @@
...
+
+/***************************************************************************************************************************************************/
+/**END OF USER CONFIGURATION ***********************************************************************************************************************/
+/***************************************************************************************************************************************************/
+/***************************************************************************************************************************************************/
 
-// DEFINES
 // Define pins used by Teensy Audio Shield
+#ifdef TEENSY_41
+#define SDCARD_CS_PIN    BUILTIN_SDCARD
```
Comment (Brototype) [inline audio-guestbook.ino:64]: if I use a teensy 4.1 with the same sd card I get a very "jumpy" recording. it skips audio pieces every couple of seconds. if is comment that line and put the sd card into the audio shield it works fine. can you help me to understand what might happen here?

**Review by Brototype — COMMENTED** [review]
thanks a lot for this - I tried it out but I have problems with recording to the sd card straight out of the teensy 4.1

**Comment by alastaira** [comment]
Hi, thanks for this.

While I agree it's in some ways convenient, my reservation about being able to record the greeting on the device itself is that it is possible for guests to accidentally (or deliberately!) over-record the greeting, and I can't really think of a scenario where you'd need to change the greeting "in the field" - can't it just be something recorded in advance and uploaded when the phone is built?

**Comment by DD4WH** [comment]
> Hi, thanks for this.
> 
> While I agree it's in some ways convenient, my reservation about being able to record the greeting on the device itself is that it is possible for guests to accidentally (or deliberately!) over-record the greeting, and I can't really think of a scenario where you'd need to change the greeting "in the field" - can't it just be something recorded in advance and uploaded when the phone is built?

No, thats a misunderstanding of the function. You can only record a greeting message, if there is no greeting.wav on the SD card. So, no accidental or deliberate overwrite of the greeting message is possible, once a greeting message is recorded or a prerecorded greeting.wav is present on the SD card. 

The primary intention of building this was motivated by the observation that MANY builders were unable to realize that they had to record a greeting message in 44.1ksps sample rate. Every greeting message recorded in other sample rates would be refused to be played back on the Teensy. So I decided to add the function to record the greeting message on the Teensy itself in order to ensure that the greeting.wav would then be playable by the Teensy.

**Comment by jvallano** [comment]
> > Hi, thanks for this.
> > While I agree it's in some ways convenient, my reservation about being able to record the greeting on the device itself is that it is possible for guests to accidentally (or deliberately!) over-record the greeting, and I can't really think of a scenario where you'd need to change the greeting "in the field" - can't it just be something recorded in advance and uploaded when the phone is built?
> 
> No, thats a misunderstanding of the function. You can only record a greeting message, if there is no greeting.wav on the SD card. So, no accidental or deliberate overwrite of the greeting message is possible, once a greeting message is recorded or a prerecorded greeting.wav is present on the SD card.
> 
> The primary intention of building this was motivated by the observation that MANY builders were unable to realize that they had to record a greeting message in 44.1ksps sample rate. Every greeting message recorded in other sample rates would be refused to be played back on the Teensy. So I decided to add the function to record the greeting message on the Teensy itself in order to ensure that the greeting.wav would then be playable by the Teensy.

Would there be a way to initiate a re-record of the greeting if the user messes up or stumbles during this process?

**Pull request closed without merging**